pipelines/docker-build-oci-ta/patch.yaml

```
@@ -0,0 +1,122 @@
+---
+- op: replace
+  path: /metadata/name
+  value: docker-build-oci-ta
+- op: replace
+  path: /metadata/labels
+  value:
+    "pipelines.openshift.io/used-by": "build-cloud"
+    "pipelines.openshift.io/runtime": "generic"
+    "pipelines.openshift.io/strategy": "docker"
+- op: remove
+  path: /spec/workspaces/0
+# Order of Tasks from the base docker-build Pipeline:
+# $ kustomize build pipelines/docker-build | yq .spec.tasks.[].name | nl -v 0
+#  0  init
+#  1  clone-repository
+#  2  prefetch-dependencies
+#  3  build-container
+#  4  build-source-image
+#  5  deprecated-base-image-check
+#  6  clair-scan
+#  7  ecosystem-cert-preflight-checks
+#  8  sast-snyk-check
+#  9  clamav-scan
+# 10  sbom-json-check
+# 11  apply-tags
+
+# clone-repository Task
+- op: replace
+  path: /spec/tasks/1/taskRef/name
+  value: git-clone-oci-ta
+- op: add
+  path: /spec/tasks/1/params/-
+  value:
+    name: ociStorage
+    value: $(params.output-image).git
+- op: add
+  path: /spec/tasks/1/params/-
+  value:
+    name: ociArtifactExpiresAfter
+    value: $(params.image-expires-after)
+- op: remove
+  path: /spec/tasks/1/workspaces/0
+
+# prefetch-dependencies Task
+- op: replace
+  path: /spec/tasks/2/taskRef/name
+  value: prefetch-dependencies-oci-ta
+- op: add
+  path: /spec/tasks/2/params/-
+  value:
+    name: SOURCE_ARTIFACT
+    value: $(tasks.clone-repository.results.SOURCE_ARTIFACT)
+- op: add
+  path: /spec/tasks/2/params/-
+  value:
+    name: ociStorage
+    value: $(params.output-image).prefetch
+- op: add
+  path: /spec/tasks/2/params/-
+  value:
+    name: ociArtifactExpiresAfter
+    value: $(params.image-expires-after)
+- op: remove
+  path: /spec/tasks/2/workspaces/0
+
+# build-container
+- op: replace
+  path: /spec/tasks/3/taskRef/name
+  value: buildah-oci-ta
+- op: add
+  path: /spec/tasks/3/params/-
+  value:
+    name: SOURCE_ARTIFACT
+    value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
+- op: add
+  path: /spec/tasks/3/params/-
+  value:
+    name: CACHI2_ARTIFACT
+    value: $(tasks.prefetch-dependencies.results.CACHI2_ARTIFACT)
+- op: remove
+  path: /spec/tasks/3/workspaces/0
+
+# build-source-image
+- op: replace
+  path: /spec/tasks/4/taskRef/name
+  value: source-build-oci-ta
+- op: add
+  path: /spec/tasks/4/params/-
+  value:
+    name: SOURCE_ARTIFACT
+    value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
+- op: add
+  path: /spec/tasks/4/params/-
+  value:
+    name: CACHI2_ARTIFACT
+    value: $(tasks.prefetch-dependencies.results.CACHI2_ARTIFACT)
+- op: remove
+  path: /spec/tasks/4/workspaces/0
```

Did you consider removing the whole thing when you expect it to be empty afterwards?
```suggestion
  path: /spec/tasks/2/workspaces
```

Just an idea for tidier yaml - I don't think it would make any functional difference.

---

I'm gonna leave it as is for consistency. The empty array is not a big deal.